exampleCourse/questions/element/matching/question.html

```diff
@@ -37,8 +37,8 @@
     </pl-question-panel>
         
     <pl-matching answers-name="ex2" counter-type="full-text">
-        <pl-statement match="Matt"><pl-figure alt="headshot of Matt" file-name="matt.png" width="140" directory="clientFilesQuestion"></pl-figure></pl-statement>
-        <pl-statement match="Mariana"><pl-figure alt="headshot of Mariana" file-name="mariana.png" width="140" directory="clientFilesQuestion"></pl-figure></pl-statement>
+        <pl-statement match="Matt" aria-label="Headshot of Matt"><pl-figure alt="headshot of Matt" file-name="matt.png" width="140" directory="clientFilesQuestion"></pl-figure></pl-statement>
+        <pl-statement match="Mariana" aria-label="Headshot of Mariana"><pl-figure alt="headshot of Mariana" file-name="mariana.png" width="140" directory="clientFilesQuestion"></pl-figure></pl-statement>
```

Comment:
We're now dealing with multiple sets of alt text. Will that be confusing to screen readers? I'm wondering if this is just a bad example; we have one further down the page that still uses images, just without the images also being the "statement".

(Complete nit: putting the names in alt text kind of destroys the point of the question, right?)

Reply:
This is a good point -- if you set `counter-type="full-text"` then the `aria-labelled-by` attribute matches that of the correct answer.

Reply:
![CleanShot 2025-05-13 at 11 20 12@2x](https://github.com/user-attachments/assets/de48d31a-4eff-4336-9c7d-29a1e20c140d)

In this example, are the images supposed to by the choices, or additional context>

Reply:
The images are "here's a thing, pick the statement that matches them".

A better example might be passages of text and their authors:

```
| Shakespeare ▼ |       To be, or not to be, that is the question
| Charles Dickens ▼ |   It was the best of times, it was the worst of times
```

Reply:
In this case, the `aria-labelledby` for `counter-type="full-text"` attribute for this scenario gives away the answer (introduced by https://github.com/PrairieLearn/PrairieLearn/pull/11844).

Reply:
This is a little funky to me -- the aria-label should probably be `headshot of X` for each `pl-statement`. Then it would make sense to put the aria-label on `pl-matching` itself, as they should all be the same.

Reply:
That's what you have already, right? My concern was the duplication of labels across `aria-label` and `alt`, and just more broadly questioning whether this is actually a good example.

Reply:
> That's what you have already, right?

Nope -- the attribute is set via `pl-statement`.

Reply:
It seems the issue is coming up with an example of a statement that would benefit from `aria-label` but that would not have an `aria-label` of its own, right? In that case, statements that are purely text don't typically need `aria-label`. Images have `alt` instead. You could use other elements like `pl-graph` or `pl-code` as the statements, but in those cases it may make more sense to implement `aria-label` in the elements themselves.

What if the documentation includes a suggestion to add the label to the content of the statement itself? As a fallback for the very limited number of cases where this is not possible, you could always have `<pl-statement ...><span aria-label="...">...</span></pl-statement>`, making this implementation not really necessary. Thoughts?

Reply:
The exact complaint from axe was that the select was labeled by something without text content, which in this case is an image. So we still need some mechanism to turn off the `aria-labellelby` attribute on the `select` element associated with each `pl-statement`. I like the idea of associating the label with the content of the statement, but I’m not sure you can say that “this option is aria-labelled by this element, which has this fallback aria text”. If that is possible, that feels like the direction to go in.

Reply:
I think we just need one level of chaining -- `aria-labelledby` -> `aria-label`. Do you know if that is allowed?

Reply:
It seems the chaining restriction only applies to `aria-labelledby`, not `aria-label` or other labeling cases. I just tested with NVDA in this same question, and it read the image's alt as the label for the select. In other words, if NVDA's case is generic enough, we can assume that the image's alt is enough for accessibility, and axe may be a bit overly restrictive.

Reply:
I also tested adding an `aria-label` to the image, and the select used the aria-label of the image as the label for the select element, so this should be enough.

Reply:
Sounds like the correct fix is to duplicate the `alt` field to the `aria-label` field of the image, and see if that resolves the Axe false positive, and that otherwise this feature isn't needed.

Reply:
Regardless of what axe reports back, you shouldn't need to duplicate alt with aria-label. This is an axe problem.

Reply:
https://gitlab.com/html-validate/html-validate/-/blame/master/src/rules/helper/has-accessible-name.ts#L96 Sorry, I looked at the error again, it is in HTMLValidate. It looks like that means that it only looks at the `div`, not the inner content (https://github.com/PrairieLearn/PrairieLearn/blob/master/apps/prairielearn/elements/pl-figure/pl-figure.mustache). According to https://w3c.github.io/accname/#computation-steps, it should be looking at child nodes as well, which it doesn't.

I can fix this in #11837 . Is there any action for this PR? I would be fine closing it out given I can workaround this.

Reply:
I don't believe there's any action for this PR, but I'd like to re-emphasize that accessibility is about the user experience, not what any given tool does or does not say. If you've tested all the existing permutations of this element and are happy with how the screen readers announces things, I'm happy to close this.

Reply:
> I'd like to re-emphasize that accessibility is about the user experience, not what any given tool does or does not say. 

Agreed. Before @jonatanschroeder 's look, I had thought that it wasn't being announced correctly, but it seems like it is, and the tooling is wrong. As long as you have some form of accessible text in the associated text for the `select`, this should just work AFAIK. I haven't tested that claim. @jonatanschroeder if you feel good that this isn't an issue in practice, I can close this out.

Reply:
Added as an issue in #11990.